python/ray/serve/scripts.py

```diff
@@ -402,22 +438,84 @@ def run(
     type=str,
     help=RAY_DASHBOARD_ADDRESS_HELP_STR,
 )
-def config(address: str):
-
-    app_info = ServeSubmissionClient(address).get_info()
-    if app_info is not None:
-        print(yaml.safe_dump(app_info, sort_keys=False))
+@click.option(
+    "--all-apps",
+    "-a",
+    is_flag=True,
+    help=(
+        "Only applies to multi-application mode. If set, shows the current app configs "
+        "of all live applications on the Ray Cluster."
+    ),
+)
+@click.option(
+    "--app-name",
+    "-n",
+    required=False,
+    type=str,
+    help=(
+        "Name of an application. Only applies to multi-application mode. If set, this "
+        "will only fetch the config for the specified app."
+    ),
+)
+def config(address: str, all_apps: bool, app_name: Optional[str]):
+    # Backwards compatible single-app behavior: displays the config for default app "".
+    if not all_apps and app_name is None:
+        app_info = ServeSubmissionClient(address).get_info()
+        if app_info is not None:
+            print(yaml.safe_dump(app_info, sort_keys=False))
+    # Multi-app support
+    else:
+        if all_apps and app_name is not None:
+            cli_logger.error("Cannot set both `--all-apps` and `--app-name`.")
+            return
+
+        serve_details = ServeSubmissionClient(address).get_serve_details()
+        if serve_details is None:
+            print("There is no Serve instance running.")
+            return
+
+        # Fetch app configs for all live applications on the cluster
+        if all_apps:
+            for app in serve_details.applications.values():
+                print("---\n")
+                print(
+                    yaml.safe_dump(
+                        app.deployed_app_config.dict(exclude_unset=True),
+                        sort_keys=False,
+                    )
+                )
+
+        # Fetch a specific app config by name.
+        elif app_name is not None:
+            if app_name not in serve_details.applications:
+                cli_logger.error(f'Application "{app_name}" does not exist.')
+            else:
+                print(
+                    yaml.safe_dump(
+                        serve_details.applications.get(
```

Comment:
Could you make a method in `ServeInstanceDetails` that returns the details for a specific app by name? We could use it here, and I think that would help with readability and reusability.

Reply:
Would this method be getting an object of type `ApplicationDetails`, or `ServeApplicationSchema`? 
- If it's `ApplicationDetails`, is the goal to be able to run `serve_details.get_app(name)` instead of `serve_details.applications.get(name)`?

Reply:
Good question– I was thinking `ServeApplicationSchema`. Do you think that logic is needed enough to warrant a separate method?

Reply:
Hmm, currently maybe not really, since it's only used in `serve config`. Perhaps in the future we can add it if more use cases come up?

Reply:
Yeah that sounds good. I'm okay with leaving this as-is. I'll resolve this conversation for now.